pkg/specgen/generate/container.go

```diff
@@ -481,7 +489,68 @@ func ConfigToSpec(rt *libpod.Runtime, specg *specgen.SpecGenerator, contaierID s
 		}
 	}
 	specg.OverlayVolumes = overlay
-	specg.Mounts = conf.Spec.Mounts
+	name, mounts := c.SortUserVolumes(c.Spec())
+	tempInspect, err := c.GetMounts(name, conf.ImageVolumes, mounts)
+	if err != nil {
+		return nil, nil, err
+	}
+	mountsToPass := []string{}
+	for _, val := range tempInspect {
+		mountsToPass = append(mountsToPass, fmt.Sprintf("type=%v,src=%v,target=%v,%v", val.Type, val.Source, val.Destination, strings.Join(val.Options, ",")))
```

Comment:
so the issue with passing the mounts directly (from `spec.Mounts`) is that you also pick up a large amount of mounts created further down in the process that in specgen cause errors (/proc for example). 

I also think its a good idea to re validate the mounts in case they break something else the user has passed (a new mount for example when I add that option to clone). 

This is basically recreating the `--mount` flag and making sure everything fits correctly. Any other way completely blew up or passed too much info

Reply:
I'm not really convinced that a full revalidation of the mount and its options is required; presumably, it was already in Podman and working.

Reply:
Why do we need to pull directly from the config? The inspect functions you're using have the details on how to determine what mounts were added by users; a slight alteration would be able to get you the relevant mounts from the config

Reply:
well, the `GetMounts` function returns a "fake" mount list that either way need to be translated to spec.Mounts for the specgen. The issue I was running into is that there were mount options that specgen does *not* like ex: `rbind` which get added later (I think?). That is the main purpose of the loop it is just pulling out invalid options that specgen itself will add later

idk if this is the best way but if you take out the loop we get smashed with a lot of options errors.

Reply:
I can definitely see needing to to a pass over the options, but the rest of the volume struct should work without change

Reply:
I was using the errors returned by the `Mounts` function as an indication of what options are causing issues below. Is there any other way to easily pass over these?

Reply:
Shouldn't be necessary, you can use the logic that inspect uses (which is relatively simple to duplicate) to determine which mounts to use, and pull them out of the existing spec - as opposed to reusing the existing function entirely

Reply:
Eh, maybe I'm being too picky here. @Luap99 @vrothberg Thoughts?

Reply:
whatever everyone prefers I can implement either way same result

Reply:
I am going to have to recreate: 
https://github.com/containers/podman/blob/a416fd6de40820e266e9e1e312c5f13bbb0b189f/pkg/specgenutil/volumes.go#L186-L237

for the options

Reply:
My commitment to see this done right is waning. Parsing again kind of offends my sensibilities, but if it works (and based on your tests it definitely does), I'm not really that committed to seeing it changed.

Reply:
Basically, if someone else says this looks fine, I'm good to merge

Reply:
ok. @containers/podman-maintainers  PTAL at this discussion. In the mean time I will keep messing around and see if I can implement it your way.

Reply:
I agree with @mheon I don't think we should parse again, the error handling below looks dangerous and fragile.